src/Microsoft.VisualStudio.Threading.Analyzers.Tests/VSTHRD010MainThreadUsageAnalyzerTests.cs

```diff
@@ -71,13 +55,21 @@ class Test {
     IVsSolution Method() { return null; }
 }
 ";
-            this.expect.Locations = new[] { new DiagnosticResultLocation("Test0.cs", 7, 23, 7, 34) };
-            this.VerifyCSharpDiagnostic(test, this.expect);
-            this.VerifyCSharpFix(test, fix);
+            var expected = Verify.Diagnostic(DescriptorSync).WithSpan(7, 23, 7, 34).WithArguments("IVsSolution", "Test.VerifyOnUIThread");
```

Comment:
>WithArguments("IVsSolution", "Test.VerifyOnUIThread") [](start = 84, length = 53)

I'm curious whether it's really worth it to verify the arguments, since I see it as an implementation detail of how the analyzer communicates with its code fix provider.

Reply:
Messages are verified by default, so if you don't care to validate the message you can use one of the following approaches:

1. Use `.WithMessage(null)` to disable this check.
2. Use markup syntax instead of explicit diagnostics

Reply:
Not *messages*, I'm talking about *arguments*. Your `WithArguments` suffix here isn't a message, is it? It looks like they are strings the analyzer tucks into the diagnostic as additional data for use by the code fix provider.

Reply:
The message format is determined automatically from the descriptor, but if it has placeholders the call to string.Format will fail unless arguments are provided.